main/src/ca/uwaterloo/flix/language/phase/jvm/ClassMaker.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020-2021 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase.jvm
+
+import ca.uwaterloo.flix.api.Flix
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.Opcodes._
+
+class ClassMaker(visitor: ClassWriter) {
+  private def makeField(fieldName: String, fieldType: JvmType, mod: Int): Unit = {
+    val field = visitor.visitField(mod, fieldName, fieldType.toDescriptor, null, null)
+    field.visitEnd()
+  }
+
+  def mkField(fieldName: String, fieldType: JvmType): Unit = {
+    makeField(fieldName, fieldType, ACC_PUBLIC)
+  }
+
+  def mkStaticField(fieldName: String, fieldType: JvmType): Unit = {
+    makeField(fieldName, fieldType, ACC_STATIC + ACC_FINAL + ACC_PUBLIC)
+  }
+
+  def mkPublicConstructor(f: BytecodeInstructions.Instruction, descriptor: String): Unit = {
```

Comment:
Why take a string? Should it not be some richer structure?

Reply:
Yes I think that would be nicer but There is no construct for method descriptors like the types have in JvmType. Everything just uses AsmOps.getMethodOf(..) or something like that. I want to add jvm strings to the types such that they are constructed once per type almost. But putting the Type `(I)Z` on Arrow(Int, Bool) seems unintuitive sometimes when you expect the interface name of the class you can call the function on.

Reply:
Sorry, I don't follow.

Reply:
Everything uses the AsmOps.getMethodDescriptor which just returns the string, so there is no datastructure used to store the strings. I've added MethodDescriptor to JvmName.